Realm/Tests/MigrationTests.mm

```diff
@@ -102,6 +103,8 @@ - (void)testGetSchemaVersion {
         [RLMRealm defaultRealm];
     }
 
+    [RLMRealm resetRealmState];
```

Comment:
we shouldn't have to call this - better way to clear the cache is to do previous operations in an autoreleasepool. This ensures the RLMRealm objects have actually been disposed of.